storagenode/piecestore/endpoint.go

```diff
@@ -315,6 +338,24 @@ func (endpoint *Endpoint) Upload(stream pb.DRPCPiecestore_UploadStream) (err err
 	defer commitOrderToStore(ctx, &largestOrder)
 
 	for {
+		// Measure upload rate (bytes per second)
+		dt := time.Since(startTime)
+
+		// Skip flagging unusually slow connections in the first moments of uploading.
+		if dt > endpoint.config.DelayUntilSlowConnectionFlagged {
+			// currentUploadSize indicates the size of uploaded data until this moment.
+			currentUploadSize := float64(pieceWriter.Size())
+
+			// averageUploadRate is counted as average rate of upload over the lifetime of
+			// the connection.
+			averageUploadRate := currentUploadSize / dt.Seconds()
```

Comment:
`dt` can be zero here

Reply:
An explicit check is added in `isSlowUpload` function